### PR TITLE
Add simplified header test for Identity

### LINKED ISSUE #975: Add some basic tests for the simplified identity header that customers use as a sanity check.
Similar to https://github.com/Azure/azure-sdk-for-cpp/pull/824, add some basic tests for the simplified identity header.

cc @antkmsft

### PR DESCRIPTION
**Pull request by antkmsft**
Closes #975.

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: sdk/identity/azure-identity/test/simplified_header.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ */
+
+#include <azure/identity.hpp>
+#include <gtest/gtest.h>
+
+class DllExportTest {
+  AZ_IDENTITY_DLLEXPORT static const bool DllExportHIncluded;
+};
+
```
Comment (ahsonkhan) [inline sdk/identity/azure-identity/test/simplified_header.cpp:17]: Why is this needed?

**Review by ahsonkhan — APPROVED** [review]

File: sdk/identity/azure-identity/test/simplified_header.cpp (hunk 1 above)
Comment (antkmsft) [reply sdk/identity/azure-identity/test/simplified_header.cpp:17]: This is to verify that `dll_import_export.hpp` is included (=macro `AZ_IDENTITY_DLLEXPORT` is defined and can syntactically be used in the context it is supposed to be used).
